### PR TITLE
Load and change profile picture

### PR DESCRIPTION
**Pull request by Morphclue**
# Context
This branch implements the new image requirement .
To realize this a new table was implemented and connected with the user.
Additionally an image is fetched for the profile view.

TODO:
- [x] Upload image
- [x] Get timestamp
- [x] Compare with database, otherwise fetch

### CODE REVIEW

File: app/src/main/java/org/feature/fox/coffee_counter/data/local/database/dao/UserDao.kt [hunk 1]
```diff
@@ -41,6 +50,10 @@ interface UserDao {
     @Query("SELECT * FROM purchase WHERE userId=:id")
     suspend fun getPurchaseListOfUser(id: String): List<Purchase>
 
+    @Transaction
+    @Query("SELECT * FROM image WHERE userId=:id")
+    suspend fun getImageById(id: String): Image
+
```
Comment (wasanderes94) [inline app/src/main/java/org/feature/fox/coffee_counter/data/local/database/dao/UserDao.kt:56]: Is there any reason, why you're using the `@Transaction` annotation?

File: app/src/main/java/org/feature/fox/coffee_counter/data/local/database/tables/Image.kt [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    )]
+)
+data class Image(
+    @PrimaryKey
+    val userId: String,
+
+    var encodedImage: String,
+
+    var timestamp: Long,
+)
```
Comment (wasanderes94) [inline app/src/main/java/org/feature/fox/coffee_counter/data/local/database/tables/Image.kt:24]: This should be an extra kotlin file for consistency reasons.

**Review by wasanderes94 — COMMENTED** [review]
Great work.

File: app/src/main/java/org/feature/fox/coffee_counter/data/local/database/dao/UserDao.kt (hunk 1 above)
Comment (Morphclue) [reply app/src/main/java/org/feature/fox/coffee_counter/data/local/database/dao/UserDao.kt]: Solved in [2af8ace](https://github.com/Morphclue/code-camp-2/pull/98/commits/2af8ace824426ffb1bfa9ba262cda5582e958247).

File: app/src/main/java/org/feature/fox/coffee_counter/data/local/database/tables/Image.kt (hunk 2 above)
Comment (Morphclue) [reply app/src/main/java/org/feature/fox/coffee_counter/data/local/database/tables/Image.kt:24]: Solved, as discussed in Discord.

**Comment by Morphclue** [comment]
This branch is currently waiting for the re-implementation of the database and its features.

File: app/src/main/java/org/feature/fox/coffee_counter/data/local/database/dao/UserDao.kt [hunk 3]
```diff
@@ -21,6 +24,9 @@ interface UserDao {
     @Update
     suspend fun insertPurchase(purchase: Purchase)
 
+    @Insert(onConflict = OnConflictStrategy.REPLACE)
+    suspend fun insertImage(image: Image)
+
```
Comment (wasanderes94) [inline app/src/main/java/org/feature/fox/coffee_counter/data/local/database/dao/UserDao.kt:29]: Pls add Db at the end of the method name to stay consistent and provide better Overview

File: app/src/main/java/org/feature/fox/coffee_counter/data/local/database/dao/UserDao.kt [hunk 4]
```diff
@@ -30,6 +36,9 @@ interface UserDao {
     @Delete
     suspend fun deletePurchase(purchase: Purchase)
 
+    @Delete
+    suspend fun deleteImage(image: Image)
+
```
Comment (wasanderes94) [inline app/src/main/java/org/feature/fox/coffee_counter/data/local/database/dao/UserDao.kt:41]: Pls add Db at the end of the method name to stay consistent and provide better Overview

File: app/src/main/java/org/feature/fox/coffee_counter/data/local/database/dao/UserDao.kt [hunk 5]
```diff
@@ -41,6 +50,9 @@ interface UserDao {
     @Query("SELECT * FROM purchase WHERE userId=:id")
     suspend fun getPurchaseListOfUser(id: String): List<Purchase>
 
+    @Query("SELECT * FROM image WHERE userId=:id")
+    suspend fun getImageById(id: String): Image?
+
```
Comment (wasanderes94) [inline app/src/main/java/org/feature/fox/coffee_counter/data/local/database/dao/UserDao.kt:55]: Pls add Db at the end of the method name to stay consistent and provide better Overview

File: app/src/main/java/org/feature/fox/coffee_counter/data/repository/UserRepository.kt [hunk 6]
```diff
@@ -46,6 +55,10 @@ class UserRepository @Inject constructor(
         userDao.deletePurchase(purchase)
     }
 
+    override suspend fun deleteImage(image: Image) {
+        userDao.deleteImage(image)
+    }
+
```
Comment (wasanderes94) [inline app/src/main/java/org/feature/fox/coffee_counter/data/repository/UserRepository.kt:61]: Pls add Db at the end of the method name to stay consistent and provide better Overview

File: app/src/main/java/org/feature/fox/coffee_counter/data/repository/UserRepository.kt [hunk 7]
```diff
@@ -58,6 +71,10 @@ class UserRepository @Inject constructor(
         return userDao.getPurchaseListOfUser(id)
     }
 
+    override suspend fun getImageByIdFromUser(id: String): Image? {
+        return userDao.getImageById(id)
+    }
+
```
Comment (wasanderes94) [inline app/src/main/java/org/feature/fox/coffee_counter/data/repository/UserRepository.kt:77]: Pls add Db at the end of the method name to stay consistent and provide better Overview

File: app/src/main/java/org/feature/fox/coffee_counter/data/repository/UserRepositoryInt.kt [hunk 8]
```diff
@@ -21,18 +23,24 @@ interface UserRepositoryInt {
...
 
+    suspend fun insertImage(image: Image)
+
     suspend fun deleteUser(user: User)
 
     suspend fun deleteFunding(funding: Funding)
 
     suspend fun deletePurchase(purchase: Purchase)
 
+    suspend fun deleteImage(image: Image)
```
Comment (wasanderes94) [inline app/src/main/java/org/feature/fox/coffee_counter/data/repository/UserRepositoryInt.kt:34]: Pls add Db at the end of the method name to stay consistent and provide better Overview

**Review by wasanderes94 — CHANGES_REQUESTED** [review]
Looks good so far. Just rework the `Image` Table as in #106 . This should also fix your missing `ForeignKey`

File: app/src/main/java/org/feature/fox/coffee_counter/data/local/database/tables/Image.kt (hunk 2 above)
[the remark this answers is not in the document]
Comment (Morphclue) [reply app/src/main/java/org/feature/fox/coffee_counter/data/local/database/tables/Image.kt]: Should be resolved in [b9b1768](https://github.com/Morphclue/code-camp-2/pull/98/commits/b9b17688b0833d7f23e68436bbee35991501e0d0).

**Review by wasanderes94 — APPROVED** [review]

**Pull request merged**